src/types.jl

```diff
@@ -97,13 +52,416 @@ for Model in [:NeuralNetworkClassifier, :ImageClassifier]
             return model
         end
 
-        @doc doc_classifier($Model) $Model
-
     end
     eval(ex)
 
 end
 
+"""
+$(MMI.doc_header(NeuralNetworkClassifier))
+
+`NeuralNetworkClassifier`: a neural network model for making probabilistic predictions
+of a Multiclass or OrderedFactor target, given a table of Continuous features. )
+ TODO:
+
+# Training data
+
+In MLJ or MLJBase, bind an instance `model` to data with
+    mach = machine(model, X, y)
+
+Where
+
+- `X`: is any table of input features (eg, a `DataFrame`) whose columns
+  are of scitype `Continuous`; check the scitype with `schema(X)`
+- `y`: is the target, which can be any `AbstractVector` whose element
+  scitype is `Multiclass` or `OrderedFactor` with `n_out` classes;
+  check the scitype with `scitype(y)`
+
+
+# Hyper-parameters
+
+- `builder=MLJFlux.Short()`: An MLJFlux builder that constructs a neural network. Possible `builders` include: `Linear`, `Short`, and `MLP`. You can construct your own builder using the `@builder` macro, see examples for further information.
+- `optimiser::Flux.ADAM()`: A `Flux.Optimise` optimiser. The optimiser performs the updating of the weights of the network. For further reference, see either the examples or [the Flux optimiser documentation](https://fluxml.ai/Flux.jl/stable/training/optimisers/). To choose a learning rate (the update rate of the optimizer), a good rule of thumb is to start out at `10e-3`, and tune using powers of 10 between `1` and `1e-7`.
```

Comment:
FYI: `ADAM` has been depecated in favour of `Adam` in Flux.

```suggestion
- `optimiser::Flux.Adam()`: A `Flux.Optimise` optimiser. The optimiser performs the updating of the weights of the network. For further reference, see either the examples or [the Flux optimiser documentation](https://fluxml.ai/Flux.jl/stable/training/optimisers/). To choose a learning rate (the update rate of the optimizer), a good rule of thumb is to start out at `10e-3`, and tune using powers of 10 between `1` and `1e-7`.
```